source/fundamentals/builders/indexes.txt

```diff
@@ -11,3 +11,165 @@ Indexes Builders
    :class: singlecol
 
 .. _indexes-builders:
+
+Overview 
+--------
+
+This guide shows you how to specify **indexes** using the Java Driver
+:doc:`builders </fundamentals/builders/>`. Index builders provide
+convenient methods for different types of indexes:
+
+- :ref:`Ascending <index-ascending>`
+- :ref:`Descending <index-descending>`
+- :ref:`Compound Indexes <index-compound-indexes>`
+- :ref:`Text Indexes <index-text-indexes>`
+- :ref:`Hashed Indexes <index-hashed-indexes>`
+- :ref:`Geospatial Indexes <index-geospatial-indexes>`
+
+Indexes store small portions of the collection’s data set. The index
+stores the value of a specific field or set of fields, ordered by the
+value of the field. See the :doc:`Indexes </fundamentals/indexes>` page
+to see examples of queries on indexes. 
+
+The :java-core-api:`Indexes <com/mongodb/client/model/Indexes.html>`
+class provides static factory methods for all the MongoDB index types.
+Each method returns a :ref:`Bson <bson>` instance, which you can pass to
+:java-sync-api:`createIndex() <com/mongodb/client/MongoCollection.html#createIndex(org.bson.conversions.Bson)>`.
+
+.. tip::
+
+   For brevity, you may choose to import all methods of the
+   :java-core-api:`Indexes <com/mongodb/client/model/Indexes.html>`
+   class statically: 
+
+   .. code-block:: java
+
+      import static com.mongodb.client.model.Indexes.*;
+
+   The examples below assume this static import.
+
+.. _index-ascending:
+
+Ascending
+---------
+
+Create a BSON instance by passing the field name or names you want to
+index to the :java-core-api:`ascending() <com/mongodb/client/model/Indexes.html#ascending(java.lang.String...)>`
+method. Then, pass the BSON instance into the createIndex() method to
+create an ascending index. 
+
+An ascending index sorts the index key from smallest to largest.
```

Comment:
Should this be at the top of the section so the reader understands what they're looking at before reading how to implement it? 

Same question for all instances.

Reply:
The index key is the name of the field, e.g.
```
{ item: 1, quantity: -1 }
```
In the above document that specifies an index, "item" and "quantity" are the index keys. The index sorts documents according to the values they hold in those fields. The index is a btree data structure that's optimized for locating and ordering results based on the type of index. MongoDB traverses the btree (index) to locate documents or provide a sort order. Documents in the collection do not contain index keys, just fields and values. 

```suggestion
An ascending index enables you to sort query results by the value of the indexed field(s) from smallest to largest.
```

Similar comment applies to all instances.